go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1200,6 +1205,54 @@ func (call *builtinMonthName) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.VarChar, Col: col, Flag: arg.Flag | flagNullable}, nil
 }
 
+func lastDay(dt datetime.DateTime) (datetime.Date, bool) {
+	ts := dt.Date.ToStdTime(time.Local)
+	firstDayOfNextMonth := time.Date(ts.Year(), ts.Month()+1, 1, 0, 0, 0, 0, time.Local)
+	lastDayOfMonth := firstDayOfNextMonth.AddDate(0, 0, -1)
```

Comment:
I think we should do the additional with `ts.AddDate(0, 1, -1)` as well and not do that in `time.Date` to make it easier to follow and to avoid the question of "how does it work if the month is already December".

Reply:
got it. done.